app/policies/rule_result_policy.rb

```diff
@@ -13,7 +13,8 @@ def show?
   # Only show RuleResults belonging to hosts visible by the current user
   class Scope < ::ApplicationPolicy::Scope
     def resolve
-      scope.select { |rule_result| Pundit.policy(user, rule_result.host) }
+      available_hosts = HostPolicy::Scope.new(user, Host).resolve.pluck(:id)
```

Comment:
Instead of iterating over every `RuleResult` object to find which ones had a `host_id` owned by the current user, now I simply get all `hosts` available for the current user and search on `RuleResult` by that.

If you turn on the SQL logging and check the before/after log, the difference is huge